integration/tests/cook/test_basic.py

```diff
@@ -25,42 +25,56 @@ def setUp(self):
         util.wait_for_cook(self.cook_url)
 
     def test_basic_submit(self):
-        job_uuid, resp = util.submit_job(self.cook_url)
-        self.assertEqual(resp.status_code, 201)
+        job_executor_type = util.get_job_executor_type(self.cook_url)
+        job_uuid, resp = util.submit_job(self.cook_url, executor=job_executor_type)
+        self.assertEqual(resp.status_code, 201, msg=resp.content)
         job = util.wait_for_job(self.cook_url, job_uuid, 'completed')
         self.assertEqual('success', job['instances'][0]['status'])
         self.assertEqual(False, job['disable_mea_culpa_retries'])
         self.assertTrue(len(util.get_output_url(self.cook_url, job_uuid)) > 0)
-        if util.is_using_cook_executor(self.cook_url):
+        if job_executor_type == 'cook':
             self.assertEqual(0, job['instances'][0]['exit_code'])
             self.assertTrue(bool(job['instances'][0]['sandbox_directory']))
 
     def test_disable_mea_culpa(self):
         job_uuid, resp = util.submit_job(self.cook_url, disable_mea_culpa_retries=True)
-        self.assertEqual(201, resp.status_code)
+        self.assertEqual(201, resp.status_code, msg=resp.content)
         job = util.load_job(self.cook_url, job_uuid)
         self.assertEqual(True, job['disable_mea_culpa_retries'])
 
         job_uuid, resp = util.submit_job(self.cook_url, disable_mea_culpa_retries=False)
-        self.assertEqual(201, resp.status_code)
+        self.assertEqual(201, resp.status_code, msg=resp.content)
         job = util.load_job(self.cook_url, job_uuid)
         self.assertEqual(False, job['disable_mea_culpa_retries'])
 
+    def test_executor_flag(self):
+        job_uuid, resp = util.submit_job(self.cook_url, executor='cook')
+        self.assertEqual(201, resp.status_code, msg=resp.content)
+        job = util.load_job(self.cook_url, job_uuid)
+        self.assertEqual('cook', job['executor'])
+
+        job_uuid, resp = util.submit_job(self.cook_url, executor='mesos')
+        self.assertEqual(201, resp.status_code, msg=resp.content)
+        job = util.load_job(self.cook_url, job_uuid)
+        self.assertEqual('mesos', job['executor'])
+
```

Comment:
Can we add integration tests that cover:

- only use cook executor on the first instance of a job
- the portion logic, say portion = 0.5, start several jobs, assert that the executor was chosen as expected

Reply:
Added:
 - integration test for no cook executor on non-first instances of a job [`test_no_cook_executor_on_subsequent_instances`]
 - unit test to verify no cook executor on non-first instances of a job despite favorable coin toss

Reply:
What about integration testing that the coin toss was correct based on the uuid?

Reply:
The challenge is that the hash computation is in Clojure and it is hard to mimic using Python. I tried using the Murmur3 python implementation mmh3, but it still produced different results than Clojure's hash function (Murmur3.hashInt). Instead, I have now added assertions in the unit tests that verify that the job uuid is being used in the hash computations.

Reply:
OK